brightcove/brightcove-popup/src/page/Popup.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState, useEffect } from "react";
+
+import { ConfigObj } from "../model/config.model";
+import { VideoList } from "../model/videoList.model";
+import { PopupObject } from "../model/receiverObj.model";
+import Modal from "../components/modal";
+
+const Popup: React.FC = ()=> {
+  const [message, setMessage] = useState("");
+  const [config, setConfig] = useState<ConfigObj>();
+  const [selectedVideosList, setSelectedVideosList] = useState<VideoList[]>();
+
+  useEffect(() => {
+    if (window.opener) {
+      window.opener.postMessage(
+        { message: "get Brightcove config and selected videos", getConfig: true },
+        "*"
+      );
+      console.info("message :" + message);
+    } else {
+      window.close();
+    }
+
+    const receiveMessage = (event: PopupObject) => {
+      const { data } = event;
+      if (data.config) {
```

Comment:
destructure the `data` obj too
You can restructure the multi level objects like 
`const { data: { config, message, selectedVideos } } = event;`

Reply:
Also, rename the fields if required
`const { data: { config: dataConfig, message, selectedVideos } } = event;`

Reply:
Also, give the default values if required
`const { data: { config, message, selectedVideos: videos = [] } } = event;`

here, you need not check the type of the videos is an array or not by giving the default value `[]` to `selectedVideos`